src/BuiltInTools/dotnet-watch/HotReload/HotReloadProfileReader.cs

```diff
@@ -11,44 +11,48 @@ internal static class HotReloadProfileReader
     {
         public static HotReloadProfile InferHotReloadProfile(ProjectGraphNode projectNode, IReporter reporter)
         {
-            var queue = new Queue<ProjectGraphNode>();
-            queue.Enqueue(projectNode);
-
-            ProjectInstance? aspnetCoreProject = null;
+            if (projectNode.IsWebApp())
+            {
+                var queue = new Queue<ProjectGraphNode>();
+                queue.Enqueue(projectNode);
 
-            var visited = new HashSet<ProjectGraphNode>();
+                ProjectInstance? aspnetCoreProject = null;
 
-            while (queue.Count > 0)
-            {
-                var currentNode = queue.Dequeue();
-                var projectCapability = currentNode.ProjectInstance.GetItems("ProjectCapability");
+                var visited = new HashSet<ProjectGraphNode>();
 
-                foreach (var item in projectCapability)
+                while (queue.Count > 0)
                 {
-                    if (item.EvaluatedInclude == "AspNetCore")
-                    {
-                        aspnetCoreProject = currentNode.ProjectInstance;
-                        break;
-                    }
-                    else if (item.EvaluatedInclude == "WebAssembly")
+                    var currentNode = queue.Dequeue();
+                    var projectCapability = currentNode.ProjectInstance.GetItems("ProjectCapability");
+
+                    foreach (var item in projectCapability)
                     {
-                        // We saw a previous project that was AspNetCore. This must he a blazor hosted app.
-                        if (aspnetCoreProject is not null && aspnetCoreProject != currentNode.ProjectInstance)
+                        if (item.EvaluatedInclude == "AspNetCore")
                         {
-                            reporter.Verbose($"HotReloadProfile: BlazorHosted. {aspnetCoreProject.FullPath} references BlazorWebAssembly project {currentNode.ProjectInstance.FullPath}.", emoji: "🔥");
-                            return HotReloadProfile.BlazorHosted;
+                            aspnetCoreProject = currentNode.ProjectInstance;
+                            break;
                         }
 
-                        reporter.Verbose("HotReloadProfile: BlazorWebAssembly.", emoji: "🔥");
-                        return HotReloadProfile.BlazorWebAssembly;
+                        if (item.EvaluatedInclude == "WebAssembly")
+                        {
+                            // We saw a previous project that was AspNetCore. This must he a blazor hosted app.
```

Comment:
What if you receive the nodes in the opposite order (WebAssembly is in the list, and later AspNetCore)?

Reply:
It doesn't matter since the condition `aspnetCoreProject is not null && aspnetCoreProject != currentNode.ProjectInstance` would exclude projects that have both capabilities anyways.

Reply:
I'll follow up with adding a blazor hosted test, seems like we don't have one